src/image.rs

```diff
@@ -1184,6 +1206,87 @@ pub struct ProgressDetail {
     total: Option<u64>,
 }
 
+/// Describes query parameters for DELETE /images/{name}
+/// https://docs.docker.com/engine/api/v1.42/#tag/Image/operation/ImageDelete
+#[derive(Default, Debug)]
+pub struct DeleteOptions {
+
+    params: HashMap<&'static str, String>,
+}
+
+impl DeleteOptions {
+    /// Remove the image even if it is being used by stopped containers or has other tags
+    pub fn force(mut self) -> Self {
+        self.params.insert("force", true.to_string());
+        self
+    }
+
+    /// Do not delete untagged parent images
+    pub fn no_prune(mut self) -> Self {
+        self.params.insert("noprune", true.to_string());
+        self
+    }
+
+    /// serialize options as a string. returns None if no options are defined
+    pub fn serialize(&self) -> Option<String> {
+        if self.params.is_empty() {
+            None
+        } else {
+            Some(
+                form_urlencoded::Serializer::new(String::new())
+                    .extend_pairs(&self.params)
+                    .finish(),
+            )
+        }
+    }
+}
+
+/// Describes query parameters for POST /images/prune
+/// https://docs.docker.com/engine/api/v1.42/#tag/Image/operation/ImagePrune
+#[derive(Default, Debug)]
+pub struct PruneOptions {
+
+    params: HashMap<&'static str, Vec<String>>,
+}
+
+impl PruneOptions {
+
+    /// dangling=<boolean> When set to true (or 1), prune only unused and untagged images.
+    /// When set to false (or 0), all unused images are pruned.
+    pub fn set_dangling(mut self, dangling: bool) -> Self {
+        self.params.insert("dangling", vec![dangling.to_string()]);
+        self
+    }
+
+    /// until=<string> Prune images created before this timestamp.
+    /// The <timestamp> can be Unix timestamps, date formatted timestamps, or Go duration strings (e.g. 10m, 1h30m) computed relative to the daemon machine’s time.
+    pub fn set_until(mut self, until: String) -> Self {
+        self.params.insert("until", vec![until]);
+        self
+    }
+
+    /// label (label=<key>, label=<key>=<value>, label!=<key>, or label!=<key>=<value>) Prune images with (or without, in case label!=... is used) the specified labels.
+    pub fn set_label(mut self, label: String) -> Self {
+        self.params.insert("label", vec![label]);
```

Comment:
Is client code supposed to be able to add more than one label filter? If so, I don't think this code supports it. And maybe `add_label` is a better name for the method, to make it clearer that it's appending to the set of labels and not replacing a single label.